### ISSUE TITLE
Date and time fields accessibility

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7321]
### Description
For date and time fields in the main content editor and details sidebar:
<img width="258" alt="Screenshot of date and date/time fields." src="https://user-images.githubusercontent.com/17884198/102929590-bbdf2280-444f-11eb-9f41-e73628abbcb4.png">
<img width="129" alt="Screenshot of time fields." src="https://user-images.githubusercontent.com/17884198/102929595-be417c80-444f-11eb-9f40-0c9d9670ebce.png">
<img width="342" alt="Screenshot of Post Date and Expiry Date date/time fields in details sidebar." src="https://user-images.githubusercontent.com/17884198/102929600-c0a3d680-444f-11eb-8441-694525bb7bf0.png">

- Calendar picker doesn't seem to be keyboard-accessible.
- Inputs are not associated with labels.
- Not sure if it's clear that the x/remove icon to the right of the time field clears both the date and time.
- The x/remove icon should be in a button element rather than an anchor.
- The x/remove icon has insufficient contrast.

### Steps to reproduce
1. Create and inspect date and time fields.

### Additional info
- Craft version: 3.5.17.1
